crates/gpui/src/platform/windows/window.rs

```diff
@@ -803,8 +791,32 @@ impl WindowsWindowInner {
         LRESULT(0)
     }
 
-    fn handle_dpi_changed_msg(&self, _msg: u32, _wparam: WPARAM, _lparam: LPARAM) -> LRESULT {
-        LRESULT(1)
+    fn handle_dpi_changed_msg(&self, wparam: WPARAM, lparam: LPARAM) -> LRESULT {
+        let new_dpi = wparam.loword() as f32;
+        let scale_factor = new_dpi / USER_DEFAULT_SCREEN_DPI as f32;
+        self.window_metrics.set_scale_factor(scale_factor);
+        let rect = unsafe { &*(lparam.0 as *const RECT) };
+        let width = rect.right - rect.left;
+        let height = rect.bottom - rect.top;
+        unsafe {
+            let _ = SetWindowPos(
+                self.hwnd,
+                None,
+                rect.left,
+                rect.top,
+                width,
+                height,
+                SWP_NOZORDER | SWP_NOACTIVATE,
+            )
+            .inspect_err(|_| {
+                log::error!(
+                    "unable to set window position after dpi has changed: {}",
+                    std::io::Error::last_os_error()
+                )
+            });
```

Comment:
Instead of ignoring the result and inspecting the error I'd suggest just using `.log_err()` or something. If you want to add context to the error you could do something like this:

```suggestion
            SetWindowPos(
                self.hwnd,
                None,
                rect.left,
                rect.top,
                width,
                height,
                SWP_NOZORDER | SWP_NOACTIVATE,
            )
            .context("unable to set window position after dpi has changed")
            .log_err();

```

Reply:
I see.